web/components/contract-search.tsx

```diff
@@ -80,6 +92,105 @@ export function ContractSearch(props: {
     headerClassName,
   } = props
 
+  const [numPages, setNumPages] = useState(1)
+  const [pages, setPages] = useState<Contract[][]>([])
+  const [showTime, setShowTime] = useState<ShowTime | undefined>()
+
+  const searchParameters = useRef<SearchParameters | undefined>()
+  const requestId = useRef(0)
+
+  const performQuery = async (freshQuery?: boolean) => {
+    if (searchParameters.current == null) {
```

Comment:
I would use `undefined` instead of `null` if that's what it is, even if `==` matches both.